### PR TITLE
fix: add warning text to roles when AUTH_ROLES_SYNC_AT_LOGIN

### PR DESCRIPTION
**Pull request by dpgaspar**
### Description

Adds a warning message when creating or editing a user and `AUTH_ROLES_SYNC_AT_LOGIN` is enabled.

Screenshot 2021-05-20 at 09 51 29

cc: @cccs-rc @thesuperzapper 

### ADDITIONAL INFORMATION
- [X] Has associated issue: #1627
- [X] Is CRUD MVC related.
- [ ] Is Auth, RBAC security related.
- [ ] Changes the security db schema.
- [ ] Introduces new feature
- [ ] Removes existing feature

### CODE REVIEW

File: flask_appbuilder/security/views.py [hunk 1]
```diff
@@ -119,6 +121,15 @@ def form_post(self, form):
         flash(as_unicode(self.message), "info")
 
 
+def _roles_custom_formatter(string: str) -> str:
+    if current_app.config.get("AUTH_ROLES_SYNC_AT_LOGIN", False):
+        string += (
+            ". <b>Warning: AUTH_ROLES_SYNC_AT_LOGIN is enabled. "
+            "Changes to this field will not persiste."
```
Comment (cccs-rc) [inline flask_appbuilder/security/views.py:128]:
```suggestion
            "Changes to this field will not persist."
```
Just a typo! :) So happy to see this getting added!

Comment (dpgaspar) [reply flask_appbuilder/security/views.py]: nice catch, thks!

**Review by cccs-rc — APPROVED** [review]

File: flask_appbuilder/security/views.py [hunk 2]
```diff
@@ -119,6 +121,15 @@ def form_post(self, form):
         flash(as_unicode(self.message), "info")
 
 
+def _roles_custom_formatter(string: str) -> str:
+    if current_app.config.get("AUTH_ROLES_SYNC_AT_LOGIN", False):
+        string += (
+            ". <b>Warning: AUTH_ROLES_SYNC_AT_LOGIN is enabled. "
+            "Changes to this field will not persist."
```
Comment (thesuperzapper) [inline flask_appbuilder/security/views.py:128]: I think "Changes to this field will not persist between user logins." is more correct

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@dpgaspar should we make the colour more obvious, say red instead of black?

**Pull request merged**